glbc: change the label of the l7-lb-controller pod

yujuhong opened the pull request:
This ensures that the default http backend service doesn't include this
pod as its endpoint. This fixes #49159

MrHohn reviewed (approved):
Thanks for the fix :)

MrHohn commented:
/lgtm

yujuhong commented:
Uh..this may cause upgrade issues since the old l7-default-backend pod will be orphaned due to the selector difference. The easier fix is to change the labels for the glbc (l7-lb-controller) pod, which seems weird since the label is `k8s-app: glbc`

MrHohn commented:
> Uh..this may cause upgrade issues since the old l7-default-backend pod will be orphaned due to the selector difference. The easier fix is to change the labels for the glbc (l7-lb-controller) pod, which seems weird since the label is k8s-app: glbc

Humm..Shouldn't the old l7-default-backend pod being cleaned up by deployment controller if we update the deployment?

MrHohn commented:
Oh just realized that updating `default-http-backend` service may hit another issue :(
https://github.com/kubernetes/ingress/issues/48

yujuhong commented:
> Humm..Shouldn't the old l7-default-backend pod being cleaned up by deployment controller if we update the deployment?

The pod will have the old label and be ignored by the new deployment.

yujuhong committed:
glbc: change the label of the l7-lb-controller pod

This ensures that the default http backend service doesn't include this
pod as its endpoint.

yujuhong commented:
> The easier fix is to change the labels for the glbc (l7-lb-controller) pod, which seems weird since the label is k8s-app: glbc

Updating the labels of the glbc pod (though ugly) seems like a more localized and safer change. Updated the PR to do so.

freehan commented:
/lgtm

MrHohn commented:
Thanks, this LGTM.

roberthbailey commented:
@yujuhong - can you update the PR description to match what the final change will be? They now seem out of sync.

yujuhong commented:
> @yujuhong - can you update the PR description to match what the final change will be? They now seem out of sync.

Done.

roberthbailey commented:
/lgtm
/approve
